### PR TITLE
grc-qt: Some improvements to the block parameter editor

### PR DESCRIPTION
**Pull request by dl1ksv**
## Description

While grc --gtk maps the dtype file_open, file_save to a filedialog, that can be used to select a file, grc --qt maps this type to a string. So the value has to be known and can't be selected by an dialog.

This is changed by this pr.

In addition the editor displays error messages at the bottom corresponding to this block, if they exist. This is already done in grc --gtk.

## Which blocks/areas does this affect?

PropsDialog
## Testing Done

Just add a yaml_config block or  some File blocks.
## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [ ] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Comment by dl1ksv** [comment]
I think the failing python formating checks does not depend on my changes

File: grc/gui_qt/components/dialogs.py [hunk 1]
```diff
@@ -170,10 +181,30 @@ def open_chooser(widget=None):
         self.buttonBox.rejected.connect(self.reject)
         self.layout = QVBoxLayout()
         self.layout.addWidget(self.tabs)
+        error_msg.setText('\n\n'.join(self._block.get_error_messages()))
+        self.layout.addWidget(error_msg)
         self.layout.addWidget(self.buttonBox)
 
         self.setLayout(self.layout)
 
+    def find_param_widget(self,button):
```
Comment (marcusmueller) [inline grc/gui_qt/components/dialogs.py:190]: needs a space after the comma :)

File: grc/gui_qt/components/dialogs.py [hunk 2]
```diff
@@ -170,10 +181,30 @@ def open_chooser(widget=None):
...
+        # Find the correct layout
+        layout = self.tabs.currentWidget().layout()
+        # Find the location in the layout of the clicked button
+        location = layout.getItemPosition(layout.indexOf(button))
+        # The required widget is at column 1
+        return layout.itemAtPosition(location[0], 1).widget()
+
+    def open_filero(self):
+        f_name, fltr = QFileDialog.getOpenFileName(self,"Open a Data File")
```
Comment (marcusmueller) [inline grc/gui_qt/components/dialogs.py:199]: `, `

File: grc/gui_qt/components/dialogs.py [hunk 3]
```diff
@@ -170,10 +181,30 @@ def open_chooser(widget=None):
...
+        button = self.sender()
+        self.find_param_widget(button).setText(f_name)
+
+    def open_filerw(self):
+        f_name, fltr = QFileDialog.getSaveFileName(self,"Save a Data File")
```
Comment (marcusmueller) [inline grc/gui_qt/components/dialogs.py:204]: `, `

**Review by marcusmueller — CHANGES_REQUESTED** [review]

File: grc/gui_qt/components/dialogs.py (hunk 2 above)
Comment (dl1ksv) [reply grc/gui_qt/components/dialogs.py:199]: Thanks for your hints. Sorry I was blind. But strange, before uploading I checked with autopep8.
But autopep proposed to insert space between " and the O. Strange.
I'll go to fix it.

**Comment by dl1ksv** [comment]
Should be fixed now.

**Pull request merged**